test/functional/sqlsrv/sqlsrv_param_output_variants.phpt

```diff
@@ -46,86 +65,68 @@ function TestOutputParam($conn, $tableName)
-            echo "initialized data and results should be the same";
-        }
-        echo "\n";
+    $stmt = execProcedure($conn, "{ CALL [$procName] ($callArgs)}", $params);
+    if (!$stmt) {
+        verifyErrorMessage(sqlsrv_errors()[0]);
     }
+    
+    dropProc($conn, $procName);
 }
 
-function TestInputAndOutputParam($conn, $tableName)
+function testInputAndOutputParam($conn, $tableName)
 {
-    $procName = GetTempProcName('sqlVariant');
+    $procName = getTempProcName('sqlVariant', false);
     $spArgs = "@p1 int, @p2 sql_variant OUTPUT";
-    $spCode = "SET @p2 = ( SELECT [c2_variant] FROM $tableName WHERE [c1_int] = @p1 )";
+    $spCode = "SET @p2 = ( SELECT [c2_variant] FROM $tableName WHERE [c1_int] = @p1)";
     $stmt = sqlsrv_query($conn, "CREATE PROC [$procName] ($spArgs) AS BEGIN $spCode END");
     sqlsrv_free_stmt($stmt);
 
     $initData = "A short text";
     $callResult = $initData;
     $phpType = SQLSRV_PHPTYPE_STRING(SQLSRV_ENC_CHAR);
 
-    $params = array( array( 1, SQLSRV_PARAM_IN ), array( &$callResult, SQLSRV_PARAM_OUT, $phpType ));
+    $params = array(array(1, SQLSRV_PARAM_IN), array(&$callResult, SQLSRV_PARAM_OUT, $phpType));
     $callArgs = "?, ?";
-    $stmt = sqlsrv_query($conn, "{ CALL [$procName] ($callArgs)}", $params);
-    if (! $stmt) {
-        print_errors();
-        if (strcmp($initData, $callResult)) {
-            echo "initialized data and results should be the same\n";
-        }
+    $stmt = execProcedure($conn, "{ CALL [$procName] ($callArgs)}", $params);
+    if (!$stmt) {
+        verifyErrorMessage(sqlsrv_errors()[0]);
     }
+
+    dropProc($conn, $procName);
 }
 
-function print_errors()
+function verifyErrorMessage($error)
 {
-    $errors = sqlsrv_errors();
-    $count = count($errors);
-
-    if ($count > 0) {
-        for ($i = 0; $i < $count; $i++) {
-            print($errors[$i]['message']."\n");
-        }
+    if (AE\isColEncrypted()) {
+        fatalError("With AE this should not have failed!");
+    }
+    
+    // Expect to fail only when AE is disabled
+    $expected = "Operand type clash: varchar(max) is incompatible with sql_variant";
+    if (strpos($error['message'], $expected) === false) {
+        echo $error['message'] . PHP_EOL;
+        fatalError("Expected error: $expected\n");
     }
 }
 
-function RunTest()
-{
-    startTest("sqlsrv_param_output_variants");
-    try {
-        setup();
-
-        // connect
-        $conn = connect();
+setup();
 
-        // Create a temp table that will be automatically dropped once the connection is closed
-        $tableName = GetTempTableName();
-        CreateVariantTable($conn, $tableName);
-        echo "\n";
+// connect
+$conn = AE\connect();
 
-        TestOutputParam($conn, $tableName);
-        TestInputAndOutputParam($conn, $tableName);
+// Create a test table that will be automatically dropped once the connection is closed
+$tableName = GetTempTableName('test_output_variants', false);
```

Comment:
shouldn't we use non temp table for all tests?

Reply:
no, `false` means it's not temporary
but I can modify this to clarify this